src/coreclr/jit/emitarm64.cpp

```diff
@@ -15125,353 +15935,915 @@ void emitter::emitIns_Call(EmitCallType          callType,
             assert((regpos >= 1) && (regpos <= 3));
             return (regpos == 2 ? PREDICATE_MERGE : PREDICATE_SIZED);
 
-        case IF_SVE_DG_2A:
-            return (regpos == 2 ? PREDICATE_ZERO : PREDICATE_SIZED);
+        case IF_SVE_DG_2A:
+            return (regpos == 2 ? PREDICATE_ZERO : PREDICATE_SIZED);
+
+        default:
+            break;
+    }
+
+    assert(!"Unexpected instruction format");
+    return PREDICATE_NONE;
+}
+
+/*****************************************************************************
+ *
+ *  Returns true if the SVE instruction has a LSL addr.
+ *  This is for formats that have [<Xn|SP>, <Xm>, LSL #N], [<Xn|SP>{, <Xm>, LSL #N}]
+ */
+/*static*/ bool emitter::insSveIsLslN(instruction ins, insFormat fmt)
```

Comment:
this kind of methods should really be converted to a table driven lookup.

Reply:
I have added this in one of the task of https://github.com/dotnet/runtime/issues/93095

Reply:
Makes sense to me, though we shouldn't do it in this PR. A follow-up will be better.